contrib/clojure-package/src/dev/generator.clj

```diff
@@ -56,37 +60,138 @@
 
 (defn increment-param-name [pname]
   (if-let [num-str (re-find #"-\d" pname)]
-    (str (first (clojure.string/split pname #"-")) "-" (inc (Integer/parseInt (last (clojure.string/split num-str #"-")))))
+    (str 
+     (first (clojure.string/split pname #"-"))
+     "-"
+     (inc (Integer/parseInt (last (clojure.string/split num-str #"-")))))
     (str pname "-" 1)))
 
-(defn rename-duplicate-params [params]
-  (reduce (fn [known-names n] (conj known-names (if (contains? (set known-names) n)
-                                                  (increment-param-name n)
-                                                  n)))
-          []
-          params))
+(defn rename-duplicate-params [pnames]
+  (->> (reduce
+        (fn [pname-counts n]
+          (let [rn (if (pname-counts n) (str n "-" (pname-counts n)) n)
+                inc-pname-counts (update-in pname-counts [n] (fnil inc 0))]
+            (update-in inc-pname-counts [:params] conj rn)))
+        {:params []}
+        pnames)
+       :params))
+
+(defn get-public-no-default-methods [obj]
+  (->> (r/reflect obj)
+       :members
+       (map #(into {} %))
+       (filter #(-> % :flags :public))
+       (filter #(not (re-find #"org\$apache\$mxnet" (str (:name %)))))
+       (filter #(not (re-find #"\$default" (str (:name %)))))))
+
+(defn get-public-to-gen-methods [public-to-hand-gen public-no-default]
+  (remove #(contains? (->> public-to-hand-gen
```

Comment:
should we use a a `let` binding to avoid recomputing the set over and over?
```clojure
(let [public-to-hand-gen-names
       (->> public-to-hand-gen
               (mapv :name)
               (mapv str)
               (set))] ...)
```
Also, we can probably use the fact that a set is a function
```clojure
(remove #(-> % :name str public-to-hand-gen-names) public-no-default)
```

Reply:
#nitpick 
We can even use a `xform` here I guess - not convinced it is more readable though
```clojure
(let [public-to-hand-gen-names
       (into #{} (mapv (comp str :name) public-to-hand-gen)] ...)
```

Reply:
done! agree that it looks much nicer :)